src/Agent.Listener/Configuration/PromptManager.cs

```diff
@@ -0,0 +1,139 @@
+using Microsoft.VisualStudio.Services.Agent.Util;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+
+namespace Microsoft.VisualStudio.Services.Agent.Listener.Configuration
+{
+    [ServiceLocator(Default = typeof(PromptManager))]
+    public interface IPromptManager: IAgentService
+    {
+        bool ReadBool(
+            string argName,
+            string description,
+            bool defaultValue,
+            Dictionary<String, String> args,
+            bool unattended);
+
+        string ReadValue(
+            string argName,
+            string description,
+            bool secret,
+            string defaultValue,
+            Func<String, bool> validator,
+            Dictionary<String, String> args,
+            bool unattended);
+    }
+
+    public class PromptManager : AgentService, IPromptManager
```

Comment:
Also secrets were getting traced. I fixed that, but I think we may be better off to just hook up the secret masker to the trace logs and terminal. There's other places we'll likely need to mask as well, like during command line parsing... more reason to hook up the secret masker.